### DISCUSSION TITLE
new Examples repo?

### DISCUSSION THREAD

**Discussion by kmartinez** [libvips/libvips#3033] [General]
I just realised it would help us empty the Wiki if we made a new examples repo ?
we could stick to markdown format with embedded code.
Maybe keep pyvips examples in pyvips repo? and stick to C/C++ ?

**Comment by jcupitt** [comment]
Maybe an `examples/` directory inside libvips? That'd be simple to do, and they'd be versioned with the library.

**Reply by kmartinez** [reply]
true.. but I suspect most code snippets would be much simplified? and hopefully this can grow to a hierarchy of examples from getting started to shrinking to writing partial-code etc? without adding to the main repo... just a thought!
benefit of /examples would be actual code files - Hmm sounds appealing too..

**Reply by jcupitt** [reply]
I don't like code fragments much -- you can't be sure they work, because you can't test them. I think everything should be complete and runnable code.  If we have `examples/` in the main libvips repo, we can compile and run them as part of the test suite and we can be sure they always work.

**Reply by kmartinez** [reply]
lets go with that then - if it becomes too huge (seems unlikely!) we can move them out ;-)

**Reply by jcupitt** [reply]
Shall I add the framework? Then anyone can just drop example code in.

**Reply by kmartinez** [reply]
Uh oh - the good old gcc -Wall vipsavg.c `pkg-config vips --cflags --libs` now throws up a can't find glib.h - a spook from the past ;-)

**Reply by jcupitt** [reply]
I pushed a basic meson.build to git master:

https://github.com/libvips/libvips/commit/873c90710a149514eb13fb6ab1975ef2727aab7f

So just add a `.c` in there, add it to the array in `meson.build` and when you build you should have a binary:

```
john@banana ~/GIT/libvips/build (master) $ ninja
[2/2] Linking target examples/annotate-animated
john@banana ~/GIT/libvips/build (master) $ ./examples/annotate-animated ~/pics/3198.gif[n=-1] x.webp
```

**Reply by kmartinez** [reply]
haha! lots to learn about meson! so I guess when we copy in code we need to update the header comment on compiling

**Reply by jcupitt** [reply]
It should still work I think, eg. I see:

```
john@banana ~/GIT/libvips/examples (master) $ gcc -g -Wall annotate-animated.c `pkg-config vips --cflags --libs`
john@banana ~/GIT/libvips/examples (master) $ ./a.out
.: usage: ./a.out xxx.gif[n=-1] yyy.gif
```

You probably have something messed up in your pkg-install. Try:

```
john@banana ~/GIT/libvips/examples (master) $ pkg-config --list-all
```

And make sure it finishes without errors.
